lib/tcpip/network_wrapper.cc

```diff
@@ -82,15 +109,37 @@ namespace
 	 * it to `operation`.  If the unsealing fails, return `-EINVAL`.  This is a
 	 * helper function used for operations on sockets.
 	 */
-	int with_sealed_socket(auto operation, Sealed<SealedSocket> sealedSocket)
+	int with_sealed_socket(bool                 isCloseOperation,
```

Comment:
Since this is true in only one(?) call site, maybe move it last and give it a default `false` parameter value?

Also, since it's statically known (and since we're specialising this function for all call sites anyway), maybe make it a template parameter and make the `if` that uses it `constexpr` so that we definitely remove it during codegen?

Reply:
Good idea, thanks!